python/openinference-instrumentation/src/openinference/instrumentation/__init__.py

```diff
@@ -12,11 +31,150 @@ def suppress_tracing() -> Iterator[None]:
             # No tracing will occur within this block
             ...
     """
-    try:
-        from opentelemetry.context import _SUPPRESS_INSTRUMENTATION_KEY, attach, detach, set_value
-    except ImportError:
-        yield
+
+    def __enter__(self) -> "suppress_tracing":
+        self._token = attach(set_value(_SUPPRESS_INSTRUMENTATION_KEY, True))
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback) -> None:
+        """Raise any exception triggered within the runtime context."""
+        detach(self._token)
+        return None
+
+
+class UsingAttributes:
+    def __init__(
+        self,
+        session_id: str = "",
```

Comment:
```suggestion
        self,
        *,
        session_id: str = "",
```

Making it required that these are keyword arguments will give us flexibility to change the order in the future, e.g., by inserting a keyword argument in the middle, without breaking backward compatibility for users.

Reply:
Great suggestion

Reply:
Great suggestion. I assume you mean only for `using_attributes`. Since `using_user(user_id="xxx")` is too redundant IMO